test/softmax.jl

```diff
@@ -48,8 +50,12 @@ end
         -0.930163 0.0519798 0.0549979 0.3799 -0.477112 0.437428
         0.69246 0.569494 -0.503191 -0.925947 -0.0870738 -1.0697
     ]
-    @test ∇logsoftmax(ones(size(xs)), xs) ≈ ys rtol = 1e-6
-    @test ∇softmax(ones(size(xs)), xs) ≈ zeros(size(xs)) atol = 1e-6
+    
+    y = logsoftmax(xs)
+    @test ∇logsoftmax(ones(size(xs)), xs, y) ≈ ys rtol = 1e-6
+    
+    y = softmax(xs)
+    @test ∇softmax(ones(size(xs)), xs, y) ≈ zeros(size(xs)) atol = 1e-6
```

Comment:
It would be good to test with more random inputs as well, since just using ones can hide mistakes in the implementation. Or was your thought that this is already covered in #242?

Reply:
I cannot change this since the r.h.s. is explicitly given, but I'am also indirectly testing this with random inputs down below when using `rrule_test`. I think this should be sufficient, even if not considering #242